src/lib.rs

```diff
@@ -166,25 +179,25 @@ impl HttpClient {
 ///
 /// By default the session is removed on `Drop`
 pub struct DriverSession {
-    driver: Box<Driver>,
+    driver: Box<Drop>,
     client: HttpClient,
     session_id: String,
     drop_session: bool,
 }
 
 impl DriverSession {
     /// Create a new session with the driver.
-    pub fn create_session(driver: Box<Driver>)
+    pub fn create_session<D: Drop + Sized + 'static>(url: &str, driver: D)
```

Comment:
You've broken down a `Driver` instance here into its __current__ important pieces: it supplies a URL and it implements `Drop` to support cleanup. When new facilities are added to `Driver` this method signature would also need to change. I prefer the OO style of passing an object and then all the method calls along the way can remain more stable.

I assume you made this change to try and work `Driver` not being clonable. Perhaps `create_session` should take a reference to `Driver` or `Driver` should be clonable. I haven't thought either option through yet, but perhaps they would give a neater design.

Reply:
`create_session` can't take a reference, because the thing being referred-to is being moved into the method. I think that if we made `Driver` extend `Drop`, we could just pass in the `Driver` here, and we could call `driver.url()` in the function... Or having `Driver` extend `Clone` would do the job too...

Reply:
One option is that `create_session` doesn't have to move the `Driver`.